src/main/java/frc/team2412/robot/subsystem/IndexSubsystem.java

```diff
@@ -1,11 +1,224 @@
+
+        public static Color BLUE_CARGO_COLOR = new Color(0, 0, 1);
+        public static Color RED_CARGO_COLOR = new Color(1, 0, 0);
+
+        // Index Motor Speeds
+
+        public static final double INDEX_IN_SPEED = 0.5; // will change values later
+        public static final double INDEX_OUT_SPEED = -0.5; // will also change later
+
+        // Proximity Threshold
+        public static final double PROXIMITY_THRESHOLD = 1000; // value not 1500, to be determined actual
+                                                                // value
+
+        // Index Motor States
+
+        public static enum IndexMotorState {
+            IN, OUT, STOPPED;
+        }
+
+        // Position States
+
+        public static enum IndexPositionState {
+            HAS_BALL, HAS_NO_BALL;
+        }
+
+        // The current limit
+        public static final SupplyCurrentLimitConfiguration MAX_MOTOR_CURRENT = new SupplyCurrentLimitConfiguration(
+                true, 40, 40, 500);
+
+    }
+
+    // Define Hardware
+
+    private final MultiplexedColorSensor ingestColorSensor;
+    private final MultiplexedColorSensor feederColorSensor;
+
+    private final WPI_TalonFX ingestMotor;
+    private final WPI_TalonFX feederMotor;
+
+    // States
+
+    private IndexMotorState ingestMotorState;
+    private IndexMotorState feederMotorState;
+
+    private IndexPositionState ingestBallState;
+    private IndexPositionState feederBallState;
+
+    // Constructor
+
+    public IndexSubsystem(WPI_TalonFX firstMotor, WPI_TalonFX secondMotor, MultiplexedColorSensor firstColorSensor,
+            MultiplexedColorSensor secondColorSensor) {
+        this.ingestMotor = firstMotor;
+        this.feederMotor = secondMotor;
+        this.ingestColorSensor = firstColorSensor;
+        this.feederColorSensor = secondColorSensor;
+
+        this.ingestMotor.configFactoryDefault();
+        this.feederMotor.configFactoryDefault();
+
+        this.ingestMotor.setNeutralMode(NeutralMode.Coast);
+        this.feederMotor.setNeutralMode(NeutralMode.Coast);
+
+        this.ingestMotor.configSupplyCurrentLimit(MAX_MOTOR_CURRENT);
+        this.feederMotor.configSupplyCurrentLimit(MAX_MOTOR_CURRENT);
+
+        ingestMotorStop();
+        feederMotorStop();
+
+        ingestBallState = HAS_NO_BALL;
+        feederBallState = HAS_NO_BALL;
+
+        setName("IndexSubsystem");
+    }
+
+    // Methods
+
+    /**
+     * Spins first motor inward and updates first motor state
+     */
+    public void ingestMotorIn() {
+        ingestMotor.set(INDEX_IN_SPEED);
+        ingestMotorState = IN;
+    }
+
+    /**
+     * Spins first motor outward and updates first motor state
+     */
+    public void ingestMotorOut() {
+        ingestMotor.set(INDEX_OUT_SPEED);
+        ingestMotorState = OUT;
+    }
+
+    /**
+     * Stops first motor and updates first motor state
+     */
+    public void ingestMotorStop() {
+        if (!ingestSensorHasBallIn()) {
```

Comment:
Probably not the right place to put this logic.  Whether or not it stops based on presence may or may depend on the command being executed.  If the drive team wants indexing to stop, it should stop.